client/client-lib/src/lib.rs

```diff
@@ -419,35 +419,46 @@ impl<T: AsRef<ClientConfig> + Clone> Client<T> {
 }
 
 impl Client<UserClientConfig> {
-    pub async fn fetch_gateway(&self) -> Result<LightningGateway> {
-        // fetch gateway from db
-        if let Some(gateway) = self
-            .context
+    pub async fn fetch_registered_gateways(&self) -> Result<Vec<LightningGateway>> {
+        Ok(self.context.api.fetch_gateways().await?)
+    }
+    pub async fn fetch_active_gateway(&self) -> Option<LightningGateway> {
+        self.context
             .db
             .get_value(&LightningGatewayKey)
             .expect("DB error")
-        {
-            return Ok(gateway);
+    }
+    pub async fn switch_active_gateway(
```

Comment:
The behavior on `None` should be documented. It's not really intuitive from reading the function signature.